experimental/Schemes/IdealSheaves.jl

```diff
@@ -576,6 +578,168 @@ function primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdea
   return result
 end
 
+## this should go to src/Rings/mpolyquo-localization.jl
+function minimal_primes(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdeal, <:MPolyLocalizedIdeal})
+  Q = base_ring(I)
+  R = base_ring(Q)
+  decomp = minimal_primes(saturated_ideal(I))
+  result = [ideal(Q, Q.(gens(b))) for b in decomp]
+  return result
+end
+
+
+function minimal_associated_points(I::IdealSheaf)
+  X = scheme(I)
+  OOX = OO(X)
+
+  charts_todo = copy(affine_charts(X))            ## todo-list of charts
+
+  associated_primes_temp = Vector{IdDict{AbsSpec, Ideal}}()  ## already identified components
+                                                  ## may not yet contain all relevant charts. but
+                                                  ## at least one for each identified component
+
+# run through all charts and try to match the components
+  while !iszero(length(charts_todo))
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                        ## supp(I) might not meet all components
+    components_here = minimal_primes(I(U))
+
+## run through all primes in MinAss(I(U)) and try to match them with previously found ones
+    for i in 1:length(components_here)
+      matches = match_on_intersections(X,U,components_here[i],associated_primes_temp,false)
```

Comment:
This is perfectly fine code, but if I may make an observation: When iterating like this over the indices of a container, and then the index `i` is only ever used to index into the container, I usually find it superior to directly iterate over the container. The resulting code is usually more compact and arguably easier to grasp. 

```suggestion
    for comp in components_here
      matches = match_on_intersections(X,U,comp,associated_primes_temp,false)
```